Aspire.Hosting.MongoDB requires impossible dependency: SharpCompress 1.0.0

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Regular `aspire update` process results in Aspire.Hosting.MongoDB being updated to 13.4.2, which requires impossible SharpCompress 1.0.0

Affected versions: 13.4.0 onwards.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### Aspire doctor output

_No response_

### Anything else?

@davidfowl

@davidfowl Can we have a fix for this? This basically hinders us to use Aspire in the latest version as we always get these package downgrade errors.

```
error NU1109:  AspireHost -> Aspire.Hosting.MongoDB 13.4.5 -> SharpCompress (>= 1.0.0)
```

There are two ways to fix it:
a) either the linked PR removing the transitive dependencies or
b) setting a valid package version for SharpCompress (0.49.1 as of now)

Thanks! :)

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -115,7 +115,7 @@
     <PackageVersion Include="Microsoft.FluentUI.AspNetCore.Components.Icons" Version="4.14.1" />
     <PackageVersion Include="Milvus.Client" Version="2.3.0-preview.1" /> <!-- No stable release available -->
     <PackageVersion Include="ModelContextProtocol" Version="1.3.0" />
-    <PackageVersion Include="MongoDB.Driver" Version="3.8.1" />
+    <PackageVersion Include="MongoDB.Driver" Version="3.9.0" />
     <PackageVersion Include="MongoDB.Driver.Core.Extensions.DiagnosticSources" Version="3.0.0" />
     <PackageVersion Include="MySqlConnector.DependencyInjection" Version="2.5.0" />
     <PackageVersion Include="MySqlConnector.Logging.Microsoft.Extensions.Logging" Version="2.1.0" />
@@ -187,9 +187,6 @@
     <!-- Pinned versions for Component Governance - Remove when root dependencies are updated -->
     <PackageVersion Include="Azure.Core" Version="1.55.0" />
     <PackageVersion Include="Azure.Identity" Version="1.21.0" />
-    <!-- MongoDB.Driver pulls these compression packages transitively; pin them here until its dependency metadata no longer requests CG-flagged versions. -->
-    <PackageVersion Include="SharpCompress" Version="1.0.0" />
-    <PackageVersion Include="Snappier" Version="1.3.1" />
     <!-- https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3313 -->
     <PackageVersion Include="Newtonsoft.Json" Version="13.0.4" />
   </ItemGroup>

